llama-index-core/llama_index/core/indices/property_graph/base.py

```diff
@@ -0,0 +1,371 @@
+            If not provided, `Settings.embed_model` will be used if `embed_kg_nodes=True`.
+        embed_kg_nodes (bool):
+            Whether to embed the KG nodes. Defaults to `True`.
+        callback_manager (Optional[CallbackManager]):
+            The callback manager to use.
+        transformations (Optional[List[TransformComponent]]):
+            A list of transformations to apply to the nodes before inserting them into the index.
+            These are applied prior to the `kg_extractors`.
+        storage_context (Optional[StorageContext]):
+            The storage context to use.
+        show_progress (bool):
+            Whether to show progress bars for transformations. Defaults to `False`.
+    """
+
+    index_struct_cls = IndexLPG
+
+    def __init__(
+        self,
+        nodes: Optional[Sequence[BaseNode]] = None,
+        llm: Optional[BaseLLM] = None,
+        kg_extractors: Optional[List[TransformComponent]] = None,
+        property_graph_store: Optional[PropertyGraphStore] = None,
+        # vector related params
+        vector_store: Optional[VectorStore] = None,
+        use_async: bool = True,
+        embed_model: Optional[EmbedType] = None,
+        embed_kg_nodes: bool = True,
+        # parent class params
+        callback_manager: Optional[CallbackManager] = None,
+        transformations: Optional[List[TransformComponent]] = None,
+        storage_context: Optional[StorageContext] = None,
+        show_progress: bool = False,
+        **kwargs: Any,
+    ) -> None:
+        """Init params."""
+        storage_context = storage_context or StorageContext.from_defaults(
+            property_graph_store=property_graph_store
+        )
+
+        # lazily initialize the graph store on the storage context
+        if property_graph_store is not None:
+            storage_context.property_graph_store = property_graph_store
+        else:
+            storage_context.property_graph_store = SimplePropertyGraphStore()
+
+        if vector_store is not None:
+            storage_context.vector_store = vector_store
+
+        if embed_kg_nodes and (
+            storage_context.property_graph_store.supports_vector_queries
+            or embed_kg_nodes
+        ):
+            self._embed_model = (
+                resolve_embed_model(embed_model)
+                if embed_model
+                else Settings.embed_model
+            )
+        else:
+            self._embed_model = None
+
+        self._kg_extractors = kg_extractors or [
+            SimpleLLMPathExtractor(llm=llm or Settings.llm),
+            ImplicitPathExtractor(),
+        ]
+        self._use_async = use_async
+        self._llm = llm
+
+        # if we aren't embedding kg nodes, don't use the vector store
+        if (
+            embed_kg_nodes
+            and not storage_context.property_graph_store.supports_vector_queries
+        ):
+            self.vector_store = storage_context.vector_store
```

Comment:
should this be a property? then can raise an error if vector_store is not specified

Reply:
Not sure what you mean here. If it's not specified, it will use SimpleVectorStore

Reply:
i just mean property_graph_store is a `@property` but this isn't. more of a nit